pulp_smash/tests/rpm/api_v2/test_updateinfo.py

```diff
@@ -0,0 +1,212 @@
+                         'arch': 'i686',
+                         'epoch': '0',
+                         'filename': 'libpfm-4.4.0-9.el7.i686.rpm',
+                         'name': 'libpfm',
+                         'release': '9.el7',
+                         'src': 'libpfm-4.4.0-9.el7.src.rpm',
+                         'sum': [
+                             'sha256',
+                             ('ca42a0d97fd99a195b30f9256823a46c94f632c126ab4f'
+                              'bbdd7e127641f30ee4')
+                         ],
+                         'version': '4.4.0'
+                     }
+                 ]}
+            ],
+            'references': [
+                {
+                    'href': 'https://example.com/errata/PULP-2017-1234.html',
+                    'id': 'PULP-2017:1234',
+                    'title': 'PULP-2017:1234',
+                    'type': 'self'
+                },
+            ],
+            'type': 'PULP',
+            'title': 'sample title',
+            'solution': 'sample solution',
+            'status': 'final',
+            'version': '6',  # intentionally string, not int,
+            'issued': '2015-03-05 05:42:53 UTC',
+        }
+
+        cls.errata['no_pkglist'] = {
+            'id': utils.uuid4(),
+            'description': 'this unit has no packages',
+            'type': 'PULP',
+            'title': 'no pkglist',
+            'solution': 'solution for no pkglist',
+            'status': 'final',
+            'version': '9',
+            'issued': '2015-04-05 05:42:53 UTC',
+        }
+
+        for (errata_key, errata_unit) in cls.errata.items():
+            import_key = 'import_%s' % errata_key
+            cls.tasks[import_key] = cls.import_erratum(
+                client, repo, errata_unit)
+
+    @classmethod
+    def setUpClass(cls):
+        """Publish a yum repo containing some updates."""
+        super(UpdateInfoTestCase, cls).setUpClass()
+
+        cls.tasks = {}
+        cls.errata = {}
+
+        client = api.Client(cls.cfg, api.json_handler)
+
+        # Create a repository for use by the test.
+        repo = client.post(REPOSITORY_PATH, gen_repo())
+        cls.resources.add(repo['_href'])
+
+        # add yum distributor to the repo
+        distribute = client.post(
+            urljoin(repo['_href'], 'distributors/'),
+            gen_distributor())
+
+        # import some errata
+        cls.import_updates(client, repo)
+
+        # ask for it to be published
+        client.post(
+            urljoin(repo['_href'], 'actions/publish/'),
+            {'id': distribute['id']})
+
+        repo_url = urljoin('/pulp/repos/',
+                           distribute['config']['relative_url'])
+        cls.updateinfo_tree = cls.get_repodata_xml(repo_url, 'updateinfo')
+
+    def assert_task_successful(self, task, msg):
+        """Assert that a task completed successfully.
+
+        This is more extensive than the checks built-in to the api Client,
```

Comment:
This test is actually less extensive than what is done by `pulp_smash.api._handle_202`. It does, however, make a fine *additional* check.